src/com/amazon/ion/benchmark/WriteRandomIonValues.java

```diff
@@ -42,145 +43,173 @@
+        String fileName = path.substring(path.lastIndexOf("/") + 1);
+        File file = new File(path);
+        
+        int[] expValRange = WriteRandomIonValues.rangeParser(expRange);
+        int[] coefficientDigitRange = WriteRandomIonValues.rangeParser(coefficientDigit);
+
+        try (OutputStream out = new BufferedOutputStream(new FileOutputStream(file));
+                IonWriter writer = WriteRandomIonValues.formatWriter(format, out)) {
             Random random = new Random();
-            // Target about 100MB of data. Decimals will average around 8 bytes, and we're writing 3 per iteration.
-            for (int i = 0; i < (size / 8 / 3); i++) {
-                int randomScale = random.nextInt(2048);
-                randomScale -= 1024;
-                writer.writeDecimal(BigDecimal.valueOf(random.nextDouble()).scaleByPowerOfTen(randomScale));
-                writer.writeDecimal(BigDecimal.valueOf(random.nextInt()));
-                writer.writeDecimal(BigDecimal.valueOf(random.nextFloat()).scaleByPowerOfTen(randomScale));
-            }
-        }
-        System.out.println("Finished writing decimals. Verifying.");
-        try (IonReader reader = IonReaderBuilder.standard().build(new BufferedInputStream(new FileInputStream(file)))) {
-            int i = 0;
-            while (reader.next() != null) {
-                if (reader.getType() != IonType.DECIMAL) {
-                    throw new IllegalStateException("Found non-decimal");
+            while (size > 0) {
+                int exp = random.nextInt((expValRange[1] - expValRange[0]) + 1) + expValRange[0];
+                int randDigits = random.nextInt((coefficientDigitRange[1] - coefficientDigitRange[0]) + 1) + coefficientDigitRange[0];
+
+                StringBuilder rs = new StringBuilder();
+
+                for (int digit = 0; digit < randDigits; digit++) {
+                    rs.append(random.nextInt(10));
                 }
-                Decimal value = reader.decimalValue();
-                if (i++ < 100) {
-                    System.out.println(value);
+                BigDecimal coefficient = new BigDecimal(rs.toString());
+                writer.writeDecimal(coefficient.scaleByPowerOfTen(exp));
+                // Calculate the size of current decimal, 4 bytes can store 9 digits number, and the decimal point == 1 byte, totalDigits is the digit number of the whole decimal.
+                int totalDigits;
+                if (Math.abs(exp) > Math.abs(randDigits)){
+                    totalDigits = Math.abs(exp);
+                } else {
+                    totalDigits = Math.abs(randDigits);
+                }
+
+                int currentSize;
+                if (totalDigits % 2 == 0) {
+                    currentSize = (totalDigits + 3) / 2;
+                } else {
+                    currentSize = (totalDigits + 4) / 2;
```

Comment:
A simpler way of doing this may be to flush the writer and check the size of the file after every X decimals written (don't do it after every decimal, as that would be super slow).